Tests/NIOSSLTests/ServerSNITests.swift

```diff
@@ -0,0 +1,154 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIO
+import NIOTLS
+import NIOSSL
+
+
+class ServerSNITests: XCTestCase {
+    
+    class SniServerContext {
+    
+        private var _tlsConfig : TLSConfiguration?;
+        
+        private var _ctx : NIOSSLContext?
+        
+        private(set) var SniExpectation : XCTestExpectation
+        
+        public var SSLContext : NIOSSLContext
+        {
+            get
+            {
+                return _ctx!
+            }
+        }
+        
+        private var _indicatedHostname : String?
+        
+        public var IndicatedHostname : String
```

Comment:
Rather than have the getters/setters, should we just use `public private(set)` and expose their optionality?